tensorflow/core/transforms/remapper/remapping_helper.h

```diff
@@ -0,0 +1,257 @@
+// `IsCpuCompatible` and `IsGpuCompatible` to check if the relevant fusion is
+// supported on CPU and GPU, respectively. Another reason to keep these
+// structures is to follow similar logics in current grappler-remapper.
+// TODO(intel-tf): Remove redundancies once the similar functionality is
+// achieved by tfg-remapper.
+struct ContractionBiasAdd {
+  Operation* contraction;
+  Operation* bias_add;
+};
+
+struct ContractionBiasAddActivation {
+  Operation* contraction;
+  Operation* bias_add;
+  Operation* activation;
+};
+
+struct ContractionBiasAddAdd {
+  Operation* contraction;
+  Operation* bias_add;
+  Operation* add;
+};
+
+struct ContractionBiasAddAddActivation {
+  Operation* contraction;
+  Operation* bias_add;
+  Operation* add;
+  Operation* activation;
+};
+
+// This enum class is used as a template parameter and meant for alias to tfg op
+// name.
+// TODO(intel-tf): Add more items as needed.
+enum class OpKind { Relu, Relu6, Elu, LeakyRelu, Tanh, Sigmoid };
+
+inline std::string GetTfgOpName(OpKind op_kind) {
+  switch (op_kind) {
+    case OpKind::Relu:
+      return "tfg.Relu";
+    case OpKind::Relu6:
+      return "tfg.Relu6";
+    case OpKind::Elu:
+      return "tfg.Elu";
+    case OpKind::LeakyRelu:
+      return "tfg.LeakyRelu";
+    case OpKind::Tanh:
+      return "tfg.Tanh";
+    case OpKind::Sigmoid:
+      return "tfg.Sigmoid";
+    default:
+      return "tfg.NoOp";
+  }
+}
+
+class OpPropertyHelper : public OpCatHelper {
+ public:
+  OpPropertyHelper() = default;
+  explicit OpPropertyHelper(TFGraphDialect* dialect,
+                            bool onednn_enabled = false,
+                            bool xla_auto_clustering = false)
+      : OpCatHelper(dialect),
+        is_onednn_enabled_(onednn_enabled),
+        is_xla_auto_clustering_enabled_(xla_auto_clustering) {}
+
+  bool HasControlOperandsOrResultUsers(Operation* op) const {
+    TFOp wrapper_op(op);
+    bool has_ctl_operands = !(wrapper_op.getControlOperands().empty());
+    bool has_ctl_ret_users = !(wrapper_op.controlRet().getUsers().empty());
+    if (has_ctl_operands || has_ctl_ret_users)
+      return true;
+    else
+      return false;
+  }
+
+  // This function is to be used for an operation that has at least 1
+  // non-control result.
+  bool HasAtMostOneUserOfResult0(Operation* op) const {
+    // All tfg operation has 1 control result. When the operation has at least 1
+    // non-control result, the number of results should be at least 2.
+    return op->getNumResults() > 1 &&
+           (op->getResult(0).hasOneUse() || op->getResult(0).use_empty());
+  }
+
+  bool IsContraction(Operation* op) const {
+    return dialect_->IsConv2D(op) || dialect_->IsConv3D(op) ||
+           dialect_->IsDepthwiseConv2dNative(op) || dialect_->IsMatMul(op);
+  }
+
+  bool HaveSameDataType(Operation* lhs_op, Operation* rhs_op,
+                        const StringRef& attr_name = "T") const {
```

Comment:
no need to pass by const ref, StringRef. Just `StringRef`.

Reply:
Done.